sources/PPU/PPU.hpp

```diff
@@ -580,27 +598,44 @@ namespace ComSquare::PPU
-		uint16_t getVramAddress();
+		uint16_t getVramAddress() const;
 		//! @brief Give the name of the Address register (used for debug)
 		std::string getValueName(uint24_t addr) const;
 		//! @brief Return true if the CPU is overloaded with debugging features.
 		virtual bool isDebugger() const;
-		//! @brief Allow others components to read the CGRAM (Debuggers)
+		//! @brief Allow others components to read the CGRAM
 		uint16_t cgramRead(uint16_t addr);
-		//! @brief Render a background on the screen
-		void renderBackground(int bgNumber, Vector2<int> characterSize, int bpp, bool priority);
-		//! @brief Get the correct Vram address for a gien x and y
-		uint16_t getGraphicVramAddress(int x, int y, int bg, int bpp);
-		//! @brief Draw a tile on the screen at x y pos
-		void drawBgTile(uint16_t data, Vector2<int> pos, int bg, int bpp, Vector2<int> characterSize);
-		//! @brief Get a palette from the number of the palette (0 - 7)
-		std::vector<uint16_t> getPalette(int nbPalette);
-		//! @brief Transform SNES color code BGR to uint32_t RGB
-		uint32_t getRealColor(uint16_t color);
-		//! @brief Get the color reference of a nb pixel tile
-		uint8_t getTilePixelReference(uint16_t addr, int bpp, int nb);
-		//! @brief draw a tilemap 32x32 starting at baseAddress
-		void drawBasicTileMap(uint16_t baseAddress, int bgNumber, int bpp, Vector2<int> characterSize, Vector2<int> offset);
+		//! @brief get the bpp depending of the bgNumber and the Bgmode
+		int getBPP(int bgNumber) const;
+		//! @brief Give the correct character size depending of the bgMode
+		Vector2<int> getCharacterSize(int bgNumber) const;
+		//! @brief Give the address where the tilemap starts
+		uint16_t getTileMapStartAddress(int bgNumber) const;
+		//! @brief Give the address to find the correct tileset for a given x and y
+		uint16_t getTilesetAddress(int bgNumber) const;
+		//! @brief Give the number of tilemaps to be rendered
+		Vector2<int> getBackgroundSize(int bgNumber) const;
+		//! @brief Render the Main and sub screen correctly
+		void renderMainAndSubScreen();
+		//! @brief Add a bg buffer to another buffer
+		template <std::size_t DEST_SIZE, std::size_t SRC_SIZE>
+		void add_buffer(std::array<std::array<uint32_t, DEST_SIZE>, DEST_SIZE> &bufferDest, std::array<std::array<uint32_t, SRC_SIZE>, SRC_SIZE> &bufferSrc);
+		//! @brief Add a bg to the sub and/or main screen
+		void addToMainSubScreen(Background &bg);
+		//! @brief Get the current background Mode
+		int getBgMode() const;
+		//! @brief update the Vram buffer
+		void updateVramReadBuffer();
+		//! @brief update the Vram buffer
+		Vector2<int> getBgScroll(int bgNumber) const;
+		//! @brief Allow to look the value of each write register (used by Register debugger)
+		const Registers &getWriteRegisters() const;
 	};
+
 }
+
+//! @brief Transform SNES color code BGR to uint32_t RGB
+uint32_t getRealColor(uint16_t color);
+int *get_dump_vram();
+int *get_dump_cgram();
```

Comment:
I believe this is for tests, if not rework it to not use raw pointers. If yes, place a TODO comment to remember to delete them.